vtex-sites/base.store
# fix: Scroll lock when transitioning on mobile via slide-over nav

## What's the purpose of this pull request?

After the changes from https://github.com/vtex-sites/base.store/pull/329, a bug was accidentally introduced when navigating using one of the links inside the mobile slide-over navigation: scrolling was locked because the `.no-scroll` CSS class was kept in the `#layout` element. The `.no-scroll` is added when the slide-over is opened to prevent scrolling the content on the background. This happened because the links have been replaced by Gatsby's so there isn't a full page refresh anymore.

## How does it work?

The fix works by removing the `.no-scroll` CSS class when needed.

[Before](https://sfj-2ce7b26--base.preview.vtex.app/)|[After](https://preview-344--base.preview.vtex.app/)
-|-
https://user-images.githubusercontent.com/381395/154371235-2d34df1c-4c51-47b1-9fda-5780b74ce4af.mp4|https://user-images.githubusercontent.com/381395/154371303-e37a4f7d-8689-46c4-a508-7452c895f791.mp4

## How to test it?

https://preview-344--base.preview.vtex.app/

1. On mobile viewport, visit a page like the [Home](https://preview-344--base.preview.vtex.app/).
2. Open the slide-over navigation and click on any of the collection/PLP/categories links.
3. Upon arriving on the new page, you should be able to scroll freely.

## References

- https://vtex-dev.atlassian.net/browse/FSSS-201
